backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/question/QuestionService.java

```diff
@@ -262,6 +283,12 @@ public void deleteQuizQuestion(QuizQuestion quizQuestion) {
             backoff = @Backoff(delay = 5000))
     @Transactional(isolation = Isolation.READ_COMMITTED)
     public void deleteQuestion(Question question) {
+        QuestionSubmission questionSubmission = questionSubmissionRepository.findQuestionSubmissionByQuestionId(question.getId());
```

Comment:
There is a removeQuestion and a deleteQuestion, which is confusing... is this method invoked from other services, besides removeQuestion?

Reply:
I have too asked myself the same question, but neither removeQuestion nor deleteQuestion were created by me and already existed. However, I believe deleteQuestion is used in deleteQuizQuestion, and since a previously submitted question can end up being a quiz question, it's important to also remove attached submission and possible reviews